File: controller/files/delete_item.php
<?php
namespace Core;
use Exception, Controller\Main;

if (!defined('__API_GOOSE__')) exit();

/**
 * delete file
 *
 * @var Goose|Connect $this
 */

try
{
  // check upload directories
  Util::checkDirectories();

  // check and set srl
  if (($srl = (int)($this->params['srl'] ?? 0)) <= 0)
  {
    throw new Exception(Message::make('error.notFound', 'srl'));
  }

  // connect db
  $this->model->connect();

  // check access
  $token = Main::checkAccessItem($this, (object)[
    'table' => 'files',
    'srl' => $srl,
  ]);

  /**
   * remove file
   */
  // get item
  $file = $this->model->getItem((object)[
    'table' => 'files',
    'field' => 'path',
    'where' => 'srl='.$srl,
  ]);

  // check exist file
  if (($file->data->path ?? false) && file_exists(__API_PATH__.'/'.$file->data->path))
  {
    unlink(__API_PATH__.'/'.$file->data->path);
  }

  // remove data
  $output = Main::delete($this, (object)[
    'table' => 'files',
    'srl' => $srl,
  ]);

  // set output
  if ($token) $output->_token = $token->jwt;

  // disconnect db
  $this->model->disconnect();

  // output data
  return Output::result($output);
}
catch (Exception $e)
{
  if ($this->model ?? false) $this->model->disconnect();
  return Error::result($e->getMessage(), $e->getCode());
}

File: src/endpoints/comment/delete_item.py
from src import output
from src.libs.db import DB, Table
from src.modules.verify import checking_token
from . import __types__ as types, __libs__ as comment_libs
from ..file import __libs__ as file_libs

async def delete_item(params: dict = {}, req = None, _db: DB = None, _token = None):

    # set values
    result = None
    db = _db if _db else DB().connect()

    try:
        # set params
        params = types.DeleteItem(**params)

        # checking token
        token = checking_token(req, db) if not _token else _token

        # get item
        item = db.get_item(
            table_name=Table.COMMENT.value,
            where=[ f'srl = {params.srl}' ],
        )
        if not item: raise Exception('Item not found', 204)

        # delete
        comment_libs.delete(db, params.srl)

        # set result
        result = output.success({
            'message': 'Complete delete comment.',
        }, _req=req)
    except Exception as e:
        result = output.exc(e, _req=req)
    finally:
        if not _db and db: db.disconnect()
        return result
